src/Cli/Microsoft.DotNet.Cli.Utils/RuntimeEnvironment.cs

```diff
@@ -179,10 +233,18 @@ private static Platform DetermineOSPlatform()
             {
                 return Platform.Darwin;
             }
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("FREEBSD")))
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Create(nameof(Platform.FreeBSD))))
```

Comment:
OSPlatform actually has a `FreeBSD` field. We should be able to change this
```suggestion
            if (RuntimeInformation.IsOSPlatform(OSPlatform.FreeBSD))
```

Reply:
OSPlatform.FreeBSD is only available for netcoreapp 3.0+, while SDK also builds against netstandard 2.0 and older runtimes.

Reply:
From looking at the code, your change is breaking this check.

https://github.com/dotnet/runtime/blob/f4d39134b8daefb5ab0db6750a203f980eecb4f0/src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs#L14

Has all upper case `FREEBSD`. And when we check `IsOSPlatform`, we call into `Interop.Sys.GetUnixName()`, which returns an all upper case string, and then we call `osPlatform.Equals(name)`, which then does a `StringComparison.Ordinal` comparison:

https://github.com/dotnet/runtime/blob/4dccf9a5e1127f8be192b6b9b2f102138d620f6d/src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs#L48

This library builds for 2 TFMs: net5.0 and net472. We don't support FreeBSD on `net472`, so you can just add a `#if NETCOREAPP` around this code and use `if (RuntimeInformation.IsOSPlatform(OSPlatform.FreeBSD))`. Then it won't break because of casing or any other string mistake.

Reply:
Thanks, fixed the casing for new platforms and used `OSPlatform.FreeBSD`. I was confused by the initial acquisition of several SDKs and thought we need to support older netcore TFMs:

```sh
$ sdk/build.sh
...
dotnet-install: Downloading legacy link: https://dotnetcli.azureedge.net/dotnet/Runtime/1.0.5/dotnet-ubuntu.16.04-x64.1.0.5.tar.gz
...
dotnet-install: Downloading legacy link: https://dotnetcli.azureedge.net/dotnet/Runtime/1.1.2/dotnet-ubuntu.16.04-x64.1.1.2.tar.gz
...
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Runtime/2.1.0/dotnet-runtime-2.1.0-linux-x64.tar.gz
...
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Runtime/2.2.8/dotnet-runtime-2.2.8-linux-x64.tar.gz
...
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Runtime/3.1.0/dotnet-runtime-3.1.0-linux-x64.tar.gz
...
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Runtime/5.0.0-preview.8.20325.1/dotnet-runtime-5.0.0-preview.8.20325.1-linux-x64.tar.gz
...
```

Reply:
Interestingly, https://github.com/dotnet/runtime/pull/39209 is changing the platform name comparison to `OridinalIgnoreCase`. :)

Reply:
It's still more correct to use the actual property names and not rely on strings.

Side question: Why aren't `ILLUMOS` and `SOLARIS` added to `OSPlatform`?

Reply:
> It's still more correct to use the actual property names and not rely on strings.

Agreed.

> Side question: Why aren't `ILLUMOS` and `SOLARIS` added to OSPlatform?

FCL implementation is in-progress for these platforms and the new properties will require API review once that work is completed. I deferred to this guideline: https://github.com/dotnet/runtime/issues/23530#issuecomment-329523856. e.g. FreeBSD port work was started in Mar 2015, and the public property was added in Nov 2018 for .NET Core 3.0, when most of runtime+libraries were made functional on the platform. By that example, it will take some time and effort to implement all libraries starting with the complex ones (such as System.Diagnostics.Process, System.IO.File etc.) for SunOS-like platforms.

Reply:
That guideline makes sense to me. Thanks for the link.